kube-core/src/labels.rs

```diff
@@ -0,0 +1,649 @@
+#![allow(missing_docs)]
+use core::fmt;
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::{LabelSelector, LabelSelectorRequirement};
+use serde::{Deserialize, Serialize};
+use std::{
+    cmp::PartialEq,
+    collections::{BTreeMap, BTreeSet},
+    fmt::Display,
+    iter::FromIterator,
+    option::IntoIter,
+};
+use thiserror::Error;
+
+use crate::ResourceExt;
+
+#[derive(Debug, Error)]
+#[error("failed to parse value as expression: {0}")]
+/// Indicates failure of conversion to Expression
+pub struct ParseExpressionError(pub String);
+
+// local type aliases
+type Expressions = Vec<Expression>;
+
+/// Extensions to [`ResourceExt`](crate::ResourceExt)
+/// Helper methods for resource selection based on provided Selector
+pub trait SelectorExt {
```

Comment:
Let's seal this trait initially to avoid it causing breaking changes if we need to change internals:

```suggestion
pub trait SelectorExt: private::Sealed {
```

Reply:
Does it work for blanket implementations on `ResourceExt`? Added, going to test in a sec.

Reply:
I’m not seeing this breaking anything. Also compiler typically complains when you try to implement something specific with a trait having a blanket implementation (type can be implemented in the future by upstream).